### PR TITLE
Split storage metrics as read and write

### PR DESCRIPTION
**Pull request by erkanerol**
**What does this PR?**

This PR replaces

- `kubevirt_vmi_storage_iops_total` with  `kubevirt_vmi_storage_iops_read_total` and `kubevirt_vmi_storage_iops_write_total`

- `kubevirt_vmi_storage_traffic_bytes_total`  with `kubevirt_vmi_storage_read_traffic_bytes_total` and `kubevirt_vmi_storage_write_traffic_bytes_total`

- `kubevirt_vmi_storage_times_ms_total`  with `kubevirt_vmi_storage_read_times_ms_total` and `kubevirt_vmi_storage_write_times_ms_total`

This PR adds detailed functional tests which check the labels of metrics and verifies that volume names of VMIs exist there.

**Special notes for your reviewer**:

This PR contains the work in #5204 and it resolves review issues on that PR.  

`AliasSet` is removed in this PR after this comment: https://github.com/kubevirt/kubevirt/pull/5204#discussion_r601208122

**Release note**:
```release-note
Some cleanups and small additions to the storage metrics
```

### CODE REVIEW

**Review by rmohr — CHANGES_REQUESTED** [review]
I think there is a logical error. Can you also copy the full PR description over and just mention in the section for reviewers that you took over the PR?

File: pkg/monitoring/vms/prometheus/prometheus.go
Comment (rmohr) [inline pkg/monitoring/vms/prometheus/prometheus.go:235]: Why is the `read` and `write` label removed? This makes it very hard to create sums and is an antipattern. :/

File: pkg/monitoring/vms/prometheus/prometheus.go
Comment (erkanerol) [reply pkg/monitoring/vms/prometheus/prometheus.go]: Fixed

File: pkg/monitoring/vms/prometheus/prometheus_test.go
Comment (erkanerol) [reply pkg/monitoring/vms/prometheus/prometheus_test.go:680]: Done

**Comment by rmohr** [comment]
/hold

Did a similar split for networking metrics occur? If yes, we have to consider reverting. It is correct that the alias gets selected over the name, but splitting the metrics seems to be the wrong action.

**Comment by erkanerol** [comment]
To the best of my knowledge, @rmohr  was waiting for the discussions about naming metrics. The discussions are completed now. Therefore we can merge this PR if it is not against any decision taken recently.

File: pkg/monitoring/vms/prometheus/prometheus.go
Comment (erkanerol) [reply pkg/monitoring/vms/prometheus/prometheus.go:236]: For others: After the discussions about kubevirt metrics, this splitting is approved at the moment.

**Comment by erkanerol** [comment]
> /hold
> 
> Did a similar split for networking metrics occur? If yes, we have to consider reverting. It is correct that the alias gets selected over the name, but splitting the metrics seems to be the wrong action.

@rmohr  knows the updates but it is better to put them here as well.

Basically, the community decided to revert splitting from 0.40 release and bring it into 0.41release. Revert PR is here #5415 We can merge this PR if there is no other problem.

File: pkg/monitoring/vms/prometheus/prometheus.go
Comment (rmohr) [reply pkg/monitoring/vms/prometheus/prometheus.go:236]: I was a little bit disappointed that there was not more input on the metrics doc. Like background on why they chose that split and so on. But since no one else seems to take interest in this, I am not blocking, although I can't say that I am a fan of it.

**Comment by rmohr** [comment]
/unhold

https://github.com/kubevirt/kubevirt/pull/5343#discussion_r616545504

**Comment by erkanerol** [comment]
@rmohr may i get an approval?

**Pull request merged**
